Upgrade to golang 1.11

refs: #115 (use vgo)

Nice to have.

@jvehent tested this out and got:

```console
# Pastebin KLVTmzV9
$ make testxpi 
go test -v -covermode=count -coverprofile=coverage_xpi.out go.mozilla.org/autograph/signer/xpi
=== RUN   TestStringToCOSEAlg
=== PAUSE TestStringToCOSEAlg
=== RUN   TestIntToCOSEAlg
=== PAUSE TestIntToCOSEAlg
=== RUN   TestGenerateCOSEKeyPair
=== PAUSE TestGenerateCOSEKeyPair
=== RUN   TestIsValidCOSESignatureErrs
=== PAUSE TestIsValidCOSESignatureErrs
=== RUN   TestIsValidCOSEMessageErrs
=== PAUSE TestIsValidCOSEMessageErrs
=== RUN   TestVerifyCOSESignaturesErrs
=== PAUSE TestVerifyCOSESignaturesErrs
=== RUN   TestIssueCOSESignatureErrs
=== PAUSE TestIssueCOSESignatureErrs
=== RUN   TestMakingJarManifest
--- PASS: TestMakingJarManifest (0.00s)
=== RUN   TestRepack
--- PASS: TestRepack (0.00s)
=== RUN   TestIsSignatureFile
--- PASS: TestIsSignatureFile (0.00s)
=== RUN   TestMetafileIsNameValid
--- PASS: TestMetafileIsNameValid (0.00s)
=== RUN   TestMakePKCS7ManifestValidatesMetafileName
--- PASS: TestMakePKCS7ManifestValidatesMetafileName (0.00s)
=== RUN   TestRepackJARWithMetafilesValidatesMetafileName
--- PASS: TestRepackJARWithMetafilesValidatesMetafileName (0.00s)
=== RUN   TestMakeEndEntity
=== PAUSE TestMakeEndEntity
=== RUN   TestSignFile
=== PAUSE TestSignFile
=== RUN   TestSignData
=== PAUSE TestSignData
=== RUN   TestSignDataAndVerifyWithOpenSSL
=== PAUSE TestSignDataAndVerifyWithOpenSSL
=== RUN   TestSignDataWithPKCS7VerifiesDigests
=== PAUSE TestSignDataWithPKCS7VerifiesDigests
=== RUN   TestNewFailure
=== PAUSE TestNewFailure
=== RUN   TestOptionsP7Digest
=== PAUSE TestOptionsP7Digest
=== RUN   TestNoID
=== PAUSE TestNoID
=== RUN   TestBadCOSEAlgsErrs
=== PAUSE TestBadCOSEAlgsErrs
=== RUN   TestBadPKCS7DigestErrs
=== PAUSE TestBadPKCS7DigestErrs
=== RUN   TestMarshalUnfinishedSignature
=== PAUSE TestMarshalUnfinishedSignature
=== RUN   TestMarshalEmptySignature
=== PAUSE TestMarshalEmptySignature
=== RUN   TestUnmarshalBadSignatureBase64
=== PAUSE TestUnmarshalBadSignatureBase64
=== RUN   TestUnmarshalBadSignaturePKCS7
=== PAUSE TestUnmarshalBadSignaturePKCS7
=== RUN   TestVerifyUnfinishedSignature
=== PAUSE TestVerifyUnfinishedSignature
=== RUN   TestRsaCaching
=== PAUSE TestRsaCaching
=== RUN   TestSignFileWithCOSESignatures
=== PAUSE TestSignFileWithCOSESignatures
=== CONT  TestStringToCOSEAlg
--- PASS: TestStringToCOSEAlg (0.00s)
=== CONT  TestSignFileWithCOSESignatures
=== CONT  TestNewFailure
=== CONT  TestSignFile
=== CONT  TestMarshalEmptySignature
=== CONT  TestIssueCOSESignatureErrs
=== CONT  TestSignDataWithPKCS7VerifiesDigests
=== CONT  TestIsValidCOSESignatureErrs
=== CONT  TestSignDataAndVerifyWithOpenSSL
=== CONT  TestSignData
--- PASS: TestIssueCOSESignatureErrs (0.00s)
--- FAIL: TestIsValidCOSESignatureErrs (0.00s)
    cose_test.go:222: validateCOSESignatureStructureAndGetEECertAndAlg returned 'xpi: failed to parse X509 EE certificate from COSE Signature: asn1: structure error: tags don't match (16 vs {class:1 tag:15 length:75 isCompound:false}) {optional:false explicit:false application:false private:false defaultValue:<nil> tag:<nil> stringType:0 timeType:0 set:false omitEmpty:false} certificate @2'
=== CONT  TestVerifyCOSESignaturesErrs
--- PASS: TestNewFailure (0.00s)
=== CONT  TestIsValidCOSEMessageErrs
--- FAIL: TestIsValidCOSEMessageErrs (0.00s)
    cose_test.go:313: validateCOSEMessageStructureAndGetCertsAndAlgs returned 'xpi: SignMessage Signature Protected Headers kid value 0 does not decode to a parseable X509 cert: asn1: structure error: tags don't match (16 vs {class:1 tag:14 length:111 isCompound:true}) {optional:false explicit:false application:false private:false defaultValue:<nil> tag:<nil> stringType:0 timeType:0 set:false omitEmpty:false} certificate @2'
=== CONT  TestMakeEndEntity
--- PASS: TestMarshalEmptySignature (0.07s)
=== CONT  TestGenerateCOSEKeyPair
--- PASS: TestSignDataAndVerifyWithOpenSSL (0.08s)
    xpi_test.go:161: OpenSSL PKCS7 signature verification output:
        foobarbaz1234abcdVerification successful
        
=== CONT  TestIntToCOSEAlg
--- PASS: TestIntToCOSEAlg (0.00s)
=== CONT  TestBadCOSEAlgsErrs
--- PASS: TestBadCOSEAlgsErrs (0.01s)
=== CONT  TestMarshalUnfinishedSignature
--- PASS: TestMakeEndEntity (0.14s)
=== CONT  TestBadPKCS7DigestErrs
--- FAIL: TestVerifyCOSESignaturesErrs (0.15s)
    cose_test.go:590: verifyCOSESignatures returned 'xpi: failed to verify EECert 0: x509: certificate is not valid for any names, but wanted to match a8a90aed72f6c28ac9cb723415558705.464b67e6be7d5509503eb06792f51426.addons.mozilla.org' but expected: 'xpi: failed to verify EECert 0: x509: certificate is valid for jid1-Kt2kYYgi32zPuw@jetpack, not a8a90aed72f6c28ac9cb723415558705.464b67e6be7d5509503eb06792f51426.addons.mozilla.org'
=== CONT  TestVerifyUnfinishedSignature
--- PASS: TestBadPKCS7DigestErrs (0.01s)
=== CONT  TestRsaCaching
--- PASS: TestMarshalUnfinishedSignature (0.09s)
=== CONT  TestUnmarshalBadSignaturePKCS7
--- PASS: TestUnmarshalBadSignaturePKCS7 (0.00s)
=== CONT  TestNoID
--- PASS: TestNoID (0.02s)
=== CONT  TestUnmarshalBadSignatureBase64
--- PASS: TestUnmarshalBadSignatureBase64 (0.00s)
=== CONT  TestOptionsP7Digest
--- PASS: TestOptionsP7Digest (0.02s)
--- PASS: TestVerifyUnfinishedSignature (0.09s)
--- PASS: TestSignDataWithPKCS7VerifiesDigests (0.55s)
--- PASS: TestGenerateCOSEKeyPair (0.67s)
--- PASS: TestSignFile (4.85s)
--- FAIL: TestSignFileWithCOSESignatures (6.46s)
    xpi_test.go:496: failed to verify signed file: xpi: error verifying COSE signatures for signed file: xpi: failed to verify EECert 0: x509: certificate specifies an incompatible key usage
--- PASS: TestRsaCaching (10.63s)
    xpi_test.go:451: retrieved rsa key from cache in 23.422µs
    xpi_test.go:456: generated rsa key without cache in 628.639147ms
--- PASS: TestSignData (13.99s)
FAIL
coverage: 85.2% of statements
FAIL    go.mozilla.org/autograph/signer/xpi     13.999s
make: *** [Makefile:61: testxpi] Error 1
```

which relates to:

>  Extended key usage restrictions are again checked only if they appear in the KeyUsages field of VerifyOptions, instead of always being checked. This matches the behavior of Go 1.9 and earlier. 

https://golang.org/doc/go1.11#crypto/x509

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,6 +1,6 @@
 dist: trusty
 language: go
-go: '1.10'
+go: '1.11'
 python: '3.6'
 services:
 - docker

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM golang:1.10
+FROM golang:1.11
 MAINTAINER Mozilla
 EXPOSE 8000
 

